contracts/access/sources/ownership_transfer/delayed.move

```diff
@@ -0,0 +1,216 @@
+/// Time-locked wrapper that enforces a configurable delay between scheduling and executing
+/// the transfer of a capability.
+///
+/// Owners wrap the capability and must queue a transfer via the scheduling helpers.
+/// After scheduling, the owner must wait for the deadline to pass before executing,
+/// guaranteeing an on-chain lead time for sensitive capability moves.
+///
+/// Unwrapping follows the same delayed pattern, ensuring the capability cannot be reclaimed
+/// without respecting the delay period.
+module openzeppelin_access::delayed_transfer;
+
+use sui::clock;
+use sui::dynamic_object_field as dof;
+use sui::event;
+
+const WRAPPED_FIELD: vector<u8> = b"wrapped";
+
+#[error(code = 0)]
+const ETransferAlreadyScheduled: vector<u8> = b"Transfer already scheduled.";
+#[error(code = 1)]
+const ENoPendingTransfer: vector<u8> = b"No pending transfer.";
+#[error(code = 2)]
+const EDelayNotElapsed: vector<u8> = b"Delay has not elapsed.";
+#[error(code = 3)]
+const EWrongPendingAction: vector<u8> = b"Pending action mismatch.";
+
+/// Wrapper object that delays transfers by at least `min_delay_ms` after scheduling.
+public struct DelayedTransferWrapper<phantom T: key + store> has key {
+    id: UID,
+    min_delay_ms: u64,
+    pending: option::Option<PendingTransfer>,
+}
+
+/// Snapshot of a scheduled transfer or unwrap, including the execution time.
+/// A non-existing recipient means an unwrap is scheduled.
+public struct PendingTransfer has drop, store {
+    recipient: option::Option<address>,
```

Comment:
`Option` type [is imported implicitly](https://move-book.com/move-basics/standard-library#implicit-imports).

```suggestion
    recipient: Option<address>,
```